pyo3-macros-backend/src/pymethod.rs

```diff
@@ -736,11 +737,21 @@ impl PropertyType<'_> {
     fn null_terminated_python_name(&self) -> Result<syn::LitStr> {
         match self {
             PropertyType::Descriptor {
-                field, python_name, ..
+                field,
+                python_name,
+                renaming_rule,
+                ..
             } => {
                 let name = match (python_name, &field.ident) {
                     (Some(name), _) => name.value.0.to_string(),
-                    (None, Some(field_name)) => format!("{}\0", field_name.unraw()),
+                    (None, Some(field_name)) => {
+                        let name = format!("{}\0", field_name.unraw());
+                        if let Some(rule) = renaming_rule {
+                            utils::apply_renaming_rule(*rule, &name)
```

Comment:
I am not sure anything bad will ever happen here, but since we rely on zero-termination for memory safety, I would prefer if we did the renaming first and then added the zero termination. (Which could maybe use `.push('\0')` then?)

Reply:
I wasn't sure either so I tested and nothing bad happened. I even tried with reserved Rust keywords. Changing nonetheless so it's clearer.